### PR TITLE
Modal Component Created

### PR DESCRIPTION
**Pull request by shrutasaralaya**
Madhu can u check the modal component

### CODE REVIEW

File: Lpicks-Core/src/app/core/modal/modal.component.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { componentFactoryName } from '@angular/compiler';
+
+@Component({
+  selector: 'kdi-modal',
+  templateUrl: './modal.component.html',
+  styleUrls: ['./modal.component.scss']
+})
+export class ModalComponent implements OnInit {
+  @Input() overlay: { title: string, content: string, option: [] };
```
Comment (Madhusuthanan-B) [inline Lpicks-Core/src/app/core/modal/modal.component.ts]: The overlay content should not be passed as string. Use content projection for this.

File: Lpicks-Core/src/app/core/modal/modal.component.html [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                </button>
+            </div>
+            <div class="kx-modal__body">
+                <!-- <ng-content></ng-content> -->
+                {{overlay.content}}
+            </div>
+
+            <div class="kx-modal__footer">
+                <div class="kx-btngroup kx-flex kx-flex-wrap kx-justify-content-center" role="group"
+                    *ngFor="let item of overlay?.option">
```
[the remark this answers is not in the document]
Comment (shrutasaralaya) [reply Lpicks-Core/src/app/core/modal/modal.component.html]: That is for buttons. User can pass as many buttons he want

**Comment by Madhusuthanan-B** [comment]
Rebase branch.

File: Lpicks-Core/src/app/core/header/header.component.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+export class HeaderComponent implements OnInit {
+  @Input() header: { hname: string, icon?: string };
+  dropdown = {
+    title: 'Profile Setiings',
+    items: [
+      {
+        icon: 'box',
+        link: '#',
+        name: 'My Profile',
+        sub: [
```
Comment (Madhusuthanan-B) [inline Lpicks-Core/src/app/core/header/header.component.ts]: Clean up all the commented code.

File: Lpicks-Core/src/app/core/modal/modal.component.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  ngOnInit() {
+  }
+  openModel() {
+    this.open = true;
+
+  }
+  closeModel() {
+    console.log('Close');
+    this.open = false;
+
```
Comment (Madhusuthanan-B) [inline Lpicks-Core/src/app/core/modal/modal.component.ts]: Remove console log and empty spaces

File: Lpicks-Core/src/app/demo/demo-header/demo-header.component.scss [hunk 5]
```diff
@@ -0,0 +1,88 @@
+// .main {
```
Comment (Madhusuthanan-B) [inline Lpicks-Core/src/app/demo/demo-header/demo-header.component.scss]: Again. Commented code. Please take care of these things.

File: Lpicks-Core/src/app/demo/demo-modal/demo-modal.component.scss [hunk 6]
```diff
@@ -0,0 +1,90 @@
+// .main {
```
Comment (Madhusuthanan-B) [inline Lpicks-Core/src/app/demo/demo-modal/demo-modal.component.scss]: Commented code!

File: Lpicks-Core/src/app/demo/demo-header/demo-header.component.scss (hunk 5 above)
Comment (shrutasaralaya) [reply Lpicks-Core/src/app/demo/demo-header/demo-header.component.scss]: Review comments addressed

File: Lpicks-Core/src/app/core/modal/modal.component.ts (hunk 4 above)
Comment (shrutasaralaya) [reply Lpicks-Core/src/app/core/modal/modal.component.ts]: review commentes addressed

**Pull request merged**